test/tools/unified-spec-runner/entities.ts

```diff
@@ -95,17 +104,54 @@ export type CmapEvent =
   | ConnectionCheckedInEvent
   | ConnectionPoolClearedEvent;
 export type SdamEvent = ServerDescriptionChangedEvent;
+export type LogMessage = Omit<ExpectedLogMessage, 'failureIsRedacted'>;
 
 function getClient(address) {
   return new MongoClient(`mongodb://${address}`, getEnvironmentalOptions());
 }
 
+// TODO(NODE-4813): Remove this class in favour of a simple object with a write method
+export class UnifiedLogCollector extends Writable {
+  collectedLogs: LogMessage[] = [];
+  observeLogMessages: Record<ObservableLogComponent, ObservableLogSeverity>;
+
+  constructor(observeLogMessages?: Record<ObservableLogComponent, ObservableLogSeverity>) {
+    super({ objectMode: true });
+    this.observeLogMessages =
+      observeLogMessages ?? ({} as Record<ObservableLogComponent, ObservableLogSeverity>);
+  }
+
+  _write(
+    log: LogMessage,
+    _: string,
+    callback: (e: Error | null, l: LogMessage | undefined) => void
+  ) {
+    const minLogLevel = this.observeLogMessages[log.component] ?? 'off';
+    const numericMinLogLevel = SeverityLevelMap.get(minLogLevel) ?? 0;
```

Comment:
have we considered mapping the log level `off` to `Infinity` or `-Infinity` instead and aligning our numeric values with what the spec refers to as the `Code` for the corresponding severity?